src/components/listSelection/ListSelection.js

```diff
@@ -4,14 +4,25 @@ import './ListSelection.css';
 import line from './line.svg';
 
 class ListSelection extends Component {
+	isLetter(str) {
+		return str.match(/^[a-z]$/i);
+	}
+
 	formattedItems = (() => {
 		let items = Object.entries(
 			_.groupBy(
 				_.sortBy(this.props.items, 'name'),
-				item => (isNaN(item.name[0]) ? item.name[0] : '#')
+				item =>
+					this.isLetter(item.name[0])
+						? item.name[0].toUpperCase()
+						: '#'
 			)
 		);
-		items.push(items.shift());
+
+		if (items[0][0] === '#') {
+			items.push(items.shift());
```

Comment:
The mutation is contained within the function so isn't too bad but we could make this immutable:

const hasMiscGroup = items[0][0] === '#'
if (hasMiscGroup) {
  const [miscGroup. ...groups] = items

  return groups.concat(miscGroup)
}

return items
```

Or slightly simpler but less descriptive
```
return items[0][0] === '#'
  ? items.slice(1).concat(items[0])
  : items
```